lib/livebook/evaluator.ex

```diff
@@ -134,12 +142,34 @@ defmodule Livebook.Evaluator do
           ref() | nil
         ) :: :ok
   def handle_intellisense(evaluator, send_to, ref, request, evaluation_ref \\ nil) do
-    GenServer.cast(evaluator, {:handle_intellisense, send_to, ref, request, evaluation_ref})
+    cast(evaluator, {:handle_intellisense, send_to, ref, request, evaluation_ref})
+  end
+
+  defp cast(evaluator, message) do
+    send(evaluator.pid, {:cast, evaluator.ref, message})
+    :ok
+  end
+
+  defp call(evaluator, message) do
+    call_ref = make_ref()
+    send(evaluator.pid, {:call, evaluator.ref, self(), call_ref, message})
+
+    receive do
+      {:call_reply, ^call_ref, reply} -> reply
```

Comment:
We can likely drop the leading element since the ref is unique.

```suggestion
      {^call_ref, reply} -> reply
```